### PR TITLE
[quant][graphmode][refactor] Change QParamMap to QParamVector

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33650 [not4land] debugging
* #33533 [not4land] Temporary workaround to get quantized ops
* #34855 [quant][graphmode] quantization support for prim::CallFunction
* #34346 [quant][graphmode] Add quantization support for aten::cat
* #34345 [quant][graphmode] Add prim::ListConstruct to general op handling
* #34807 [quant][graphmode] quantization support for prim::ListUnpack
* #34854 [quant][graphmode] quantization support for view, transpose, contiguos
* #34806 [quant][graphmode] quantization support for aten::chunk
* #34805 [quant][graphmode] add constant prop in finalize to simplify graph
* #34804 [quant][graphmode] Replicate quantize node for prim::If
* #34572 [quant][graphmode] quantization support for aten::add
* #34518 [quant][graphmode] quantization work for prim::If
* #34571 [quant][graphmode][fix] use observed_values_ to check values are observed
* #34414 [quant][graphmode][refactor] insertObservers for Block
* #34411 [quant][graphmode][fix] Insert dequantize before use node
* #34349 [quant][graphmode][fix] preserve the type of original value when inserting dequant node
* #34348 [quant] Add dequantize.tensors
* #34803 [quant][graphmode] quantization support for aten::rehshape
* #34347 [quant][graphmode] Add quantization support for aten::dropout
* #34315 [quant][graphmode] insert quant/dequant work for duplicated debugName
* **#34314 [quant][graphmode][refactor] Change QParamMap to QParamVector**
* #33854 [quant][graphmode] Quantization pattern for aten::linear

Summary:

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20493032](https://our.internmc.facebook.com/intern/diff/D20493032)

### CODE REVIEW

File: torch/csrc/jit/passes/quantization.cpp [hunk 1]
```diff
@@ -30,7 +30,7 @@ using NameModuleVector = std::vector<std::pair<std::string, script::Module>>;
 // Map of quantization parameter name and value
 // for example _scale, _zero_point,
 // _scalar_type and _axis(for per channel quantization)
-using QParamMap = std::unordered_map<std::string, IValue>;
+using QParamVector = std::vector<std::pair<std::string, IValue>>;
```
Comment (raghuramank100) [inline torch/csrc/jit/passes/quantization.cpp]: Can you point to a test that checks per channel quantization?

Comment (jerryzh168) [reply torch/csrc/jit/passes/quantization.cpp]: https://github.com/pytorch/pytorch/blob/master/test/test_jit.py#L1291

File: torch/csrc/jit/passes/quantization.cpp [hunk 2]
```diff
@@ -1376,19 +1381,19 @@ std::tuple<c10::QScheme, QParamMap> InsertQuantDeQuantHelper::
...
+  // quantization parameters should appear in the same order as
+  // the argument for quantize_per_tensor/quantize_per_channel function
+  QParamVector qparams;
   auto qscheme = observer_module.attr("qscheme").toQScheme();
-  if (qscheme == c10::kPerChannelAffine ||
-      qscheme == c10::kPerChannelSymmetric) {
-    qparams["_scale"] = scale;
-    qparams["_zero_point"] = zero_point;
-    qparams["_axis"] = tp->elements()[2].toInt();
+  if (isPerChannel(qscheme)) {
```
Comment (supriyar) [inline torch/csrc/jit/passes/quantization.cpp]: Should we be checking if `qscheme` is valid before this line?

**Review by supriyar — APPROVED** [review]
One comment, otherwise looks good.

File: torch/csrc/jit/passes/quantization.cpp (hunk 2 above)
Comment (jerryzh168) [reply torch/csrc/jit/passes/quantization.cpp]: I think we don't have invalid qscheme right now: https://bddppq.github.io/codebrowser/pytorch/pytorch/c10/core/QScheme.h.html

**Pull request closed without merging**